spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -56,90 +57,105 @@
 
 	private final Log log = LogFactory.getLog(ChannelBindingService.class);
 
-	private BinderFactory<MessageChannel> binderFactory;
+	private BinderFactory binderFactory;
 
 	private final ChannelBindingServiceProperties channelBindingServiceProperties;
 
-	private final Map<String, Binding<MessageChannel>> producerBindings = new HashMap<>();
+	private final Map<String, Binding<?>> producerBindings = new HashMap<>();
 
-	private final Map<String, List<Binding<MessageChannel>>> consumerBindings = new HashMap<>();
+	private final Map<String, List<Binding<?>>> consumerBindings = new HashMap<>();
 
-	public ChannelBindingService(ChannelBindingServiceProperties channelBindingServiceProperties,
-			BinderFactory<MessageChannel> binderFactory) {
+	public ChannelBindingService(
+			ChannelBindingServiceProperties channelBindingServiceProperties,
```

Comment:
I'm not sure deprecated is the right term here, as there is no replacement for it. Clearly using  it like that was a bad idea, but there's not much we can do here.